### PR TITLE
telemetry/analytics

### PR DESCRIPTION
**Pull request by disq**
- [x] Set up collector/receiver end
- [x] Hostname and connection details

### CODE REVIEW

**Comment by spangenberg** [comment]
So far I really like it :)

File: pkg/plugin/manager.go
Comment (roneli) [inline pkg/plugin/manager.go:85]: would add a method to Plugin Interface to get Provider Full name which should include details.Provider + details.Version, this is also regarding line 76

File: internal/telemetry/telemetry.go
Comment (roneli) [inline internal/telemetry/telemetry.go:256]: I would say this should be hardcoded, maybe make it configurable in the future so we can support proxies

File: internal/telemetry/telemetry.go
Comment (roneli) [inline internal/telemetry/telemetry.go:222]: document the cookie function :)

File: internal/telemetry/telemetry.go
Comment (roneli) [inline internal/telemetry/telemetry.go:40]: please document the parameters

**Review by roneli — CHANGES_REQUESTED** [review]
Looking good, added a few comments

File: pkg/plugin/manager.go
Comment (disq) [reply pkg/plugin/manager.go:85]: The only problem is `registry.ProviderDetails` doesn't seem to implement the `Plugin` interface.

File: internal/telemetry/telemetry.go
Comment (disq) [reply internal/telemetry/telemetry.go:245]: Done

File: internal/telemetry/telemetry.go
Comment (disq) [reply internal/telemetry/telemetry.go]: Done

File: cmd/util.go
Comment (roneli) [inline cmd/util.go:21]: I liked this fix, makes command much cleaner, not sure handleError is correct anymore since it does more, maybe handleCommand / executeCommand, you add the handle error inside this function, or just keep it as is in the current function.

**Review by roneli — COMMENTED** [review]
LGTM, I won't approve yet, unless you don't have any more changes, ill have another go over it to double check

File: cmd/util.go
Comment (disq) [reply cmd/util.go]: it's now `handleCommand`

File: pkg/plugin/manager.go
Comment (roneli) [reply pkg/plugin/manager.go:85]: So add a method to `registry.ProviderDetails` ?

File: cmd/telemetry.go
Comment (roneli) [inline cmd/telemetry.go:28]: maybe add a log line explaining that cq-telemetry was created locally ?

**Review by roneli — APPROVED** [review]
LGTM

File: cmd/telemetry.go
Comment (disq) [reply cmd/telemetry.go]: Done

File: pkg/plugin/manager.go
Comment (disq) [reply pkg/plugin/manager.go:85]: It has to be the whole interface, so 5/6 methods. It just reads the struct fields without utilizing any interface abstraction. I don't want to touch that in this PR.

**Review by spangenberg — APPROVED** [review]

**Pull request merged**
